src/upload.rs

```diff
@@ -1,3 +1,6 @@
+//! This module handles the expected information a crate should have
+//! and manages the serialising and deserialising of this information
```

Comment:
I think so too, because the only way I see it being separated is to duplicate these structs in `src/tests/`, which isn't ideal.